lib/Dialect/FIRRTL/LowerToRTL.cpp

```diff
@@ -133,6 +133,34 @@ static LogicalResult lower(firrtl::PadPrimOp op, ArrayRef<Value> operands,
   return success();
 }
 
+static LogicalResult lower(firrtl::CatPrimOp op, ArrayRef<Value> operands,
+                           ConversionPatternRewriter &rewriter) {
+  auto lhs = mapOperand(operands[0], op, rewriter);
+  auto rhs = mapOperand(operands[1], op, rewriter);
+  if (!lhs || !rhs)
+    return failure();
+
+  auto lhsWidth = lhs.getType().cast<IntegerType>().getWidth();
+  auto rhsWidth = rhs.getType().cast<IntegerType>().getWidth();
+
+  // TODO full argument type conversion
+  Value args [2] = {operands[0], operands[1]};
+
+  auto resultLoc = op.getLoc();
+  Type resultTypes [1] = {rewriter.getIntegerType(lhsWidth + rhsWidth)};
```

Comment:
Single values convert to array ref, so this can be:

Type resultTypes = rewriter.getIntegerType(lhsWidth + rhsWidth);

Reply:
fixed